sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CProperties.java

```diff
@@ -84,6 +95,34 @@
 
     private String replyUrl = "{baseUrl}/login/oauth2/code/";
 
+    /**
+     * AAD B2C endpoint base uri.
+     */
+    @URL(message = "baseUri should be valid URL")
+    private String baseUri;
+
+    @Override
+    public void afterPropertiesSet() {
+        if (StringUtils.isEmpty(tenant) && StringUtils.isEmpty(baseUri)) {
+            throw new AADB2CConfigurationException("'tenant' and 'baseUri' at least configure one item.");
+        }
+        // baseUri points to Global env by default
+        if (StringUtils.hasText(baseUri) && StringUtils.isEmpty(tenant)) {
+            baseUri = AADB2CURL.addSlash(baseUri);
```

Comment:
What I meant is this add slash part should be done when we construct the endpoints, not here.